### PR TITLE
Bugfix/pocketlisp/num to text

### PR DESCRIPTION
**Pull request by vektor8891**
**Issue number:** #517 

**Changelog:**
- fix bug in num-to-text
- add unit tests
- refactoring

### CODE REVIEW

File: src/shared/script/shared-code/utils.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (maxinteger) [inline src/shared/script/shared-code/utils.ts:1]: usually ESLint give us useful notes with warnings
Please do not disable it

**Review by maxinteger — CHANGES_REQUESTED** [review]

File: src/shared/script/shared-code/utils.ts (hunk 1 above)
Comment (vektor8891) [reply src/shared/script/shared-code/utils.ts:1]: @maxinteger how can i fix these warnings?
```Unexpected any. Specify a different type.eslint@typescript-eslint/no-explicit-any```
and
```Argument 'a' should be typed with a non-any type.eslint@typescript-eslint/explicit-module-boundary-type```

Comment (maxinteger) [reply src/shared/script/shared-code/utils.ts]: I just checked the file and looks like safe to use `any` in the `assertType` and `typeCheck`

please disable only for these 2 functions the eslint check:
```
// eslint-disable-next-line @typescript-eslint/no-explicit-any 
```

**Review by maxinteger — CHANGES_REQUESTED** [review]

File: src/shared/script/shared-code/utils.ts (hunk 1 above)
Comment (vektor8891) [reply src/shared/script/shared-code/utils.ts]: fixed

**Pull request merged**